src/main/kotlin/db/migration/data/dev/R__2_5_Integration_test_data.kt

```diff
@@ -0,0 +1,265 @@
+import uk.gov.justice.digital.hmpps.pecs.jpc.move.MoveStatus
+import uk.gov.justice.digital.hmpps.pecs.jpc.move.MoveType
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.Supplier
+import uk.gov.justice.digital.hmpps.pecs.jpc.price.effectiveYearForDate
+import java.time.LocalDate
+import java.util.UUID
+
+/**
+ * This is a code based migration which serves two (main) purposes:
+ *
+ * 1. Generate a fresh set of moves on the day of execution to assist with development.
+ * 2. As in point one, we can the moves in the integration tests as we have fine control over the move dates.
+ *
+ */
+@Suppress("ClassName", "unused")
+class R__2_5_Integration_test_data : BaseJavaMigration() {
+
+  override fun migrate(context: Context) {
+    JdbcTemplate(SingleConnectionDataSource(context.connection, true)).also {
+      createStandardMoves(it)
+    }
+  }
+
+  private fun createStandardMoves(template: JdbcTemplate) {
+    mapOf(
+      "SM1" to "PR1",
+      "SM2" to "PR2",
+      "SM3" to "PR3"
+    ).forEach {
+      create(move(moveId = it.key, profileId = it.value, fromAgencyId = "SFROM", toAgencyId = "STO"), template).also { move ->
+        create(moveStartEvent(move), template)
+        create(moveCompleteEvent(move), template)
+        create(journey(move, fromAgencyId = "SFROM", toAgencyId = "STO"), template).also { journey ->
+          create(journeyStartEvent(journey), template)
+          create(journeyCompleteEvent(journey), template)
+        }
+      }
+    }
+  }
+
+  private fun create(move: Move, template: JdbcTemplate): Move {
+    template.update(
+      moveSql,
+      move.dropOffOrCancelledDateTime,
+      move.fromNomisAgencyId,
+      move.moveDate,
+      move.moveId,
+      move.moveType!!.name,
+      move.notes,
+      move.pickUpDateTime,
+      move.profileId,
+      move.reference,
+      move.reportFromLocationType,
+      move.reportToLocationType,
+      move.status.name,
+      move.supplier.name,
+      move.toNomisAgencyId,
+      move.updatedAt
+    )
+
+    return move
+  }
+
+  private fun create(journey: Journey, template: JdbcTemplate): Journey {
+    template.update(
+      journeySql,
+      journey.billable,
+      journey.clientTimeStamp,
+      journey.dropOffDateTime,
+      journey.effectiveYear,
+      journey.fromNomisAgencyId,
+      journey.journeyId,
+      journey.moveId,
+      journey.notes,
+      journey.pickUpDateTime,
+      journey.state.name,
+      journey.supplier.name,
+      journey.toNomisAgencyId,
+      journey.updatedAt,
+      journey.vehicleRegistration
+    )
+
+    return journey
+  }
+
+  private fun create(event: Event, template: JdbcTemplate) {
+    template.update(
+      eventSql,
+      event.eventId,
+      event.eventableId,
+      event.eventableType,
+      event.notes,
+      event.occurredAt,
+      event.recordedAt,
+      event.supplier?.name,
+      event.type,
+      event.updatedAt
+    )
+  }
+}
+
+private val today = LocalDate.now()
```

Comment:
This is one of the key elements here.  This enables fine grained control of the dates and times of the moves, journeys and events.